ossindex/internal/cache/cache.go

```diff
@@ -34,20 +34,36 @@ import (
 // DBName can be used to override the actual database name, primarily for testing
 // TTL can be used to set the amount of time a specific object can live in the cache
 type Cache struct {
+	Options Options
+	logLady *logrus.Logger
+}
+
+// Options is a struct with values necessary to run the cache package
+type Options struct {
+	// DBDirName is the directory that will be created or used for the cache DB, defaults to nancy
+	DBDirName string
+	// DBName is the actual name of the cache database on disk
 	DBName string
-	TTL    time.Time
+	// TTL is the time each entry into the cache will live for before being replaced
+	TTL time.Time
 }
 
 const dbDirName = "nancy"
 
-var logLady *logrus.Logger
-
 // DBValue is a local struct used for adding a TTL to a Coordinates struct
 type DBValue struct {
 	Coordinates types.Coordinate
 	TTL         int64
 }
 
+// New is intended to be the way to obtain a cache instance
+func New(logger *logrus.Logger, options Options) *Cache {
+	if options.DBDirName == "" {
```

Comment:
Could this be nil??  And/or should we trim and check for empty string??

Reply:
Strings cannot be nil in Golang, that's my core understanding, they all get initialized as "", since it's really just a character array (or rune array).

I haven't done much digging but is there a `strings.IsEmpty`? If there is I wouldn't be opposed to using that!

Reply:
This is what I found digging further: https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go

WE IDIOMATIC!

Reply:
👍 dumb java dev ....acting as a golang dev 😜 

Maybe something like this just to be safe?? Don’t see a helper method in the core lib. 
`len(strings.TrimSpace(s)) == 0`

Reply:
That's reasonable!

Reply:
In: ba49717